src/lib/services/project-service.ts

```diff
@@ -269,10 +269,12 @@ export default class ProjectService {
         if (!isCompatibleWithTargetProject) {
             throw new IncompatibleProjectError(newProjectId);
         }
+        const createdBy = getCreatedBy(user);
+        if (!createdBy) return;
```

Comment:
```suggestion
```
After adding `|| 'unknown'` is it possible to get undefined here?

Reply:
Good catch. I also noticed that we have createdBy || 'system-user' in other places so I'm thinking about defaulting to 'system-user' instead of 'unknown'. Again this should probably not happen in practice but we should be defensive.

Reply:
I think we're using `'unknown'` in other cases, and I wonder if there's such need... throwing an exception might be better, although it might also uncover some hidden errors we already have. Not a problem but I'd leave it in sandbox and maybe demo instance for a while and monitor logs.

Reply:
Leaving 'unknown' for now. And agree not to change behavior. So throwing TypeError for now.